mappings/net/minecraft/client/gui/widget/TextFieldWidget.mapping

```diff
@@ -74,8 +74,8 @@ CLASS net/minecraft/class_342 net/minecraft/client/gui/widget/TextFieldWidget
 		ARG 1 newText
 	METHOD method_1875 setSelectionStart (I)V
 		ARG 1 cursor
-	METHOD method_1876 setSelected (Z)V
-		ARG 1 selected
+	METHOD method_1876 setTextFieldFocused (Z)V
```

Comment:
Hmm, having `TextField` in name doesn't sound quite good. Is this about whether the cursor for typing is in the field? If yes I'd say `setHasCursor` or alike

Reply:
I don't think `setHasCursor` is accurate, because it does more than that. The focused property in `AbstractButtonWidget` controls whether the text field controls both whether the cursor is shown and keyboad input is accepted. All this method does is call `super.setFocused`.

Reply:
hmm, what does this method do? is it an explicit override to return this? worth checking in javap, as i recall vec3i and blockpos has such up and down methods that override for a more specific return type

Reply:
It literally just calls the `AbstractButtonWidget.setFocused` because the superclass method is protected and the text field needs to make it public.

In `TextFieldWidget`:
![image](https://user-images.githubusercontent.com/7091588/108109306-a9ecbb00-705f-11eb-902d-20a606b4ade2.png)
In `AbstractButtonWidget`:
![image](https://user-images.githubusercontent.com/7091588/108109333-b244f600-705f-11eb-9740-cd6fe6c0b513.png)

Reply:
After a look at the class file it appears they are indeed distinct methods and `method_1876` is not some sort of bridge. :+1: